core/src/main/java/org/elasticsearch/common/util/CancellableThreads.java

```diff
@@ -92,6 +94,7 @@ public void execute(Interruptable interruptable) {
      *
      * @param interruptable code to run
      */
+    @SuppressForbidden(reason = "Use Thread.interrupted() to implement this class' interruption policy (forbid external interrupts).")
```

Comment:
Tbh, I am not happy at all that `@SuppressForbidden` takes an all or nothing approach. I really don't want to extract a method that makes no sense from a purely functional perspective just to satisfy `@SuppressForbidden`. So maybe we should work on improving `@SuppressForbidden`?

Ideally, I'd like to do something like:

```java
@SuppressForbidden(rule="Thread#interrupted()", reason = "Use Thread.interrupted() to implement this class' interruption policy (forbid external interrupts).")
```

This would *only* suppress this single rule but still check all others.